client/js/widgets/deck.js

```diff
@@ -21,6 +21,8 @@ class Deck extends Widget {
   }
 
   cardPropertyGet(cardType, property) {
+    if(this.cards[cardId].deck.state.faceTemplates[activeFace || 0][property] !== undefined)
```

Comment:
Where does `cardId` come from? But in general `this.cards[cardId].deck` is just `this`, isn't it? Should just be `this.p('faceTemplates')` unless I'm missing something.

It looks like for `BasicWidget` the behavior is that it doesn't default to face `0` if `activeFace` contains an invalid value. So this code should check if the face `activeFace` exists and if it doesn't, skip this value.

Reply:
`faceTemplates` is a deck property not a card property.

Reply:
this has been corrected (and moved to card.js)